### PR TITLE
feat: V2 MultiScreenPage view

### PR DESCRIPTION
**Pull request by cmaddox5**
**Asana task**: [Create all-screens pages for v2](https://app.asana.com/0/1185117109217413/1201514401502519/f)

Went ahead and did this for v2 DUPs and Solari, even though there are no v2 versions of those screens. The ticket also say not to worry about Pre-Fare, but it was super low effort to implement it for them after the general solution was implemented.

- [ ] Needs version bump?

### CODE REVIEW

File: lib/screens_web/controllers/v2/screen_controller.ex [hunk 1]
```diff
@@ -81,4 +92,34 @@ defmodule ScreensWeb.V2.ScreenController do
...
+    |> Base.encode16()
+    |> String.to_integer(16)
+    |> rem(refresh_rate)
+  end
+
+  defp screen_ids(target_app_id, refresh_rate) do
+    ids =
+      for {screen_id, %Screen{app_id: ^target_app_id}} <- State.screens() do
+        screen_id
+      end
```
Comment (PaulJKim) [inline lib/screens_web/controllers/v2/screen_controller.ex:107]: Is this for loop effectively the same as using `Enum.map` to do the same thing? If so, any reason to prefer one over the other?

File: lib/screens_web/controllers/v2/screen_controller.ex [hunk 2]
```diff
@@ -81,4 +92,34 @@ defmodule ScreensWeb.V2.ScreenController do
...
+    end)
+  end
+
+  defp id_sort_fn(a, b) do
+    case {Integer.parse(a), Integer.parse(b)} do
+      {{m, ""}, {n, ""}} ->
+        m <= n
+
+      _ ->
+        a <= b
```
[the remark this answers is not in the document]
Comment (cmaddox5) [reply lib/screens_web/controllers/v2/screen_controller.ex:122]: So while there are not currently any non-numeric IDs for V2 screens, DUPs could go to V2 somewhere down the line and their IDs follow the pattern `DUP-${location}`. With that in mind, I decided to keep this logic to account for those if they are used.

File: assets/src/hooks/v2/use_api_response.tsx [hunk 3]
```diff
@@ -96,10 +96,20 @@ const useApiResponse = ({
...
+  const {
+    lastRefresh,
+    refreshRate,
+    refreshRateOffset,
+    screenIdsWithOffsetMap,
+  } = document.getElementById("app").dataset;
   const refreshMs = parseInt(refreshRate, 10) * 1000;
-  const refreshRateOffsetMs = parseInt(refreshRateOffset, 10) * 1000;
+  let refreshRateOffsetMs = parseInt(refreshRateOffset, 10) * 1000;
+  if (!refreshRateOffsetMs && screenIdsWithOffsetMap) {
```
[the remark this answers is not in the document]
Comment (PaulJKim) [reply assets/src/hooks/v2/use_api_response.tsx]: Yep that's sort of what I was thinking might be possible in the back of my head, but I didn't think I understood the context enough. I think just checking for `screensIdWithOffsetMap` should work

File: lib/screens_web/controllers/v2/screen_controller.ex (hunk 1 above)
Comment (PaulJKim) [reply lib/screens_web/controllers/v2/screen_controller.ex:107]: Gotcha. Yeah, the anonymous function passed into the filter is a little hard to read too

**Review by PaulJKim — APPROVED** [review]
LGTM

**Pull request merged**